core/src/main/java/com/predic8/membrane/core/interceptor/authentication/BasicAuthenticationInterceptor.java

```diff
@@ -95,7 +95,10 @@ private String getAuthorizationHeaderDecoded(Exchange exc) {
 	}
 
 	public List<User> getUsers() {
-		return ((StaticUserDataProvider)userDataProvider).getUsers();
+		if (userDataProvider.getClass() == StaticUserDataProvider.class) {
```

Comment:
public to package
Use instanceof with let construct
throw not ImplementedException/Illegal? or equivallent.